unit/java_bytecode/java_bytecode_parse_generics/parse_generic_class.cpp

```diff
@@ -16,307 +18,148 @@
+            to_java_generic_parameter(field_t.type());
+          REQUIRE(field_t_param.type_variable().get_identifier() ==
+                  class_prefix + "::T");
+        }
       }
-    }
-  }
 
-  GIVEN("Some class files with generic type variable")
-  {
-    WHEN("Parsing a class with bounded type variable")
-    {
-      REQUIRE(new_symbol_table.has_symbol("java::generics"));
-
-      THEN("The generic fields should be annotated with concrete types")
+      THEN("It has field g pointing to Generic")
       {
-        const symbolt &class_symbol=
-          new_symbol_table.lookup_ref("java::generics");
-        const typet &symbol_type=class_symbol.type;
-
-        REQUIRE(symbol_type.id()==ID_struct);
-        class_typet class_type=to_class_type(symbol_type);
-        REQUIRE(class_type.is_class());
-        java_class_typet java_class_type=to_java_class_type(class_type);
-        REQUIRE(!is_java_generics_class_type(java_class_type));
+        struct_union_typet::componentt field_g =
+          require_type::require_component(class_struct, "g");
+        require_type::require_pointer(
+          field_g.type(), symbol_typet("java::Generic"));
 
-        const struct_union_typet::componentt &belem=
-          java_class_type.get_component("belem");
-        const typet &belem_type=java_class_type.component_type("belem");
-
-        REQUIRE(belem_type!=nil_typet());
-        REQUIRE(is_java_generic_type(belem_type));
-        THEN("Field has instantiated type variable")
+        THEN("It is generic with parameter pointing to java.lang.Integer")
         {
-          const java_generic_typet &container=
-            to_java_generic_type(belem_type);
-
-          const std::vector<java_generic_parametert> &generic_types=
-            container.generic_type_variables();
-          REQUIRE(generic_types.size()==1);
-
-          const typet& inst_type=java_generic_get_inst_type(0, container);
-
-          REQUIRE(inst_type.id()==ID_pointer);
-          const typet &inst_type_symbol=inst_type.subtype();
-          REQUIRE(inst_type_symbol.id()==ID_symbol);
-          REQUIRE(
-            to_symbol_type(inst_type_symbol).get_identifier()==
-            "java::java.lang.Integer");
+          REQUIRE(is_java_generic_type(field_g.type()));
+
+          const java_generic_typet generic_field_g =
+            to_java_generic_type(field_g.type());
+          const java_generic_parametert generic_param_field_g =
+            generic_field_g.generic_type_variables().front();
+          require_type::require_pointer(
+            generic_param_field_g, symbol_typet("java::java.lang.Integer"));
         }
       }
     }
   }
+}
 
-  GIVEN("Some class files with Generics")
-  {
-    WHEN("Methods with generic signatures")
-    {
-      REQUIRE(
-        new_symbol_table
-          .has_symbol("java::generics$bound_element.f:()Ljava/lang/Number;"));
+SCENARIO(
+  "java_bytecode_parse_generic_class_two_param",
```

Comment:
Ah, I see you've split the scenario into multiple scenarios. I guess there is a design question to ask around whether we want one scenario per file or not. I'm open either way, but worth considering.

Reply:
See comment above.